cli/util/git.go

```diff
@@ -17,32 +17,33 @@ func GetRepoName(dir string) (string, error) {
 		return "", fmt.Errorf("error getting remote URL: %w", err)
 	}
 
-	trimmedRemotePath := strings.TrimSuffix(remoteUrl, "/")
+	trimmedRemotePath := strings.TrimSuffix(remoteUrl.Path, "/")
 	splitRemotePath := strings.Split(trimmedRemotePath, "/")
 	// expect path to be /owner/repo
 	if len(splitRemotePath) != 3 {
 		return "", fmt.Errorf("expected owner/repo, got %s", trimmedRemotePath)
 	}
 
-	return splitRemotePath[len(splitRemotePath)-1], nil
+	repoName := strings.TrimSuffix(splitRemotePath[len(splitRemotePath)-1], ".git")
+	return repoName, nil
 }
 
 // getRepoName finds upstream repo name from a given repo directory
-func GetRepoUrl(dir string) (string, error) {
+func GetRepoUrl(dir string) (*url.URL, error) {
```

Comment:
Let's add some test cases with the earlier issue so we won't have regressions.

Reply:
added